### PR TITLE
feat: Fixed tab item idx properties

### PR DESCRIPTION
**Pull request by jonlima2018**
# Description
Fixed the item tab idx property

## Type of change
- [X] Decisive change (fix or feature that would cause existing functionality not to work as expected)

# How was this tested?
Test created including the idx property and passed.
Assigned idx in Tabs and validated in DOM.

# Checklist:
- [X] My changes do not generate new warnings

### CODE REVIEW

File: src/Tabs.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ const Tabs = ({ children, className, defaultValue, options, onChange }) => {
     <React.Fragment>
       <ul className={cx('tabs', className)} ref={_tabsRef}>
         {React.Children.map(children, (child, id) => {
-          const idx = `${scope}${id}`;
+          const idx = child.props.idx ? child.props.idx : `${scope}${id}`;
           const { active, disabled, tabWidth, title } = child.props;
```
[the remark this answers is not in the document]
Comment (jonlima2018) [reply src/Tabs.js]: i'm going to implement

File: src/Tabs.js [hunk 2]
```diff
@@ -45,7 +45,7 @@ const Tabs = ({ children, className, defaultValue, options, onChange }) => {
       </ul>
       <Row>
         {React.Children.map(children, (child, id) => {
-          const idx = `${scope}${id}`;
+          const idx = child.props.idx ? child.props.idx : `${scope}${id}`;
```
[the remark this answers is not in the document]
Comment (jonlima2018) [reply src/Tabs.js]: i'm created a new function to render the Tabs in new commit

**Review by alextrastero — COMMENTED** [review]
How about using `useMemo` to only generate this map when children change?

File: stories/Tabs.stories.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
-import React from 'react';
 import { storiesOf } from '@storybook/react';
-import Tabs from '../src/Tabs';
+import React from 'react';
 import Tab from '../src/Tab';
+import Tabs from '../src/Tabs';
 
 const stories = storiesOf('Javascript/Tabs', module);
```
Comment (alextrastero) [inline stories/Tabs.stories.js:6]: Do we need to change this file?

**Comment by alextrastero** [comment]
Overall I feel this component is more complex than it should be, the creation of a memoized component is not really necessary.

I would create a memoized func to render the tabs, no need for it for the content as it just loops and assigns an `idx`.

**Pull request closed without merging**